tests/core/test_table.py

```diff
@@ -0,0 +1,275 @@
+            _ = run.optimization.tables.create(
+                name="Table 3",
+                constrained_to_indexsets=[indexset.name, indexset.name],
+                column_names=["Column 1", "Column 1"],
+            )
+
+        # Test column.dtype is registered correctly
+        indexset_2 = run.optimization.indexsets.create("Indexset 2")
+        indexset_2.add(elements=2024)
+        table_3 = run.optimization.tables.create(
+            "Table 5",
+            constrained_to_indexsets=["Indexset", indexset_2.name],
+        )
+        # If indexset doesn't have elements, a generic dtype is registered
+        assert table_3.columns[0].dtype == "object"
+        assert table_3.columns[1].dtype == "int64"
+
+    def test_get_table(self, test_mp, request):
+        test_mp: Platform = request.getfixturevalue(test_mp)  # type: ignore
+        run = test_mp.runs.create("Model", "Scenario")
+        indexset = run.optimization.indexsets.create(name="Indexset")
+        _ = run.optimization.tables.create(
+            name="Table", constrained_to_indexsets=["Indexset"]
+        )
+        table = run.optimization.tables.get("Table")
+        assert table.run_id == run.id
+        assert table.id == 1
+        assert table.name == "Table"
+        assert table.data == {}
+        assert table.columns[0].name == indexset.name
+        assert table.constrained_to_indexsets == [indexset.name]
+
+        with pytest.raises(Table.NotFound):
+            _ = run.optimization.tables.get(name="Table 2")
+
+    def test_table_add_data(self, test_mp, request):
+        test_mp: Platform = request.getfixturevalue(test_mp)  # type: ignore
+        run = test_mp.runs.create("Model", "Scenario")
+        indexset = run.optimization.indexsets.create("Indexset")
+        indexset.add(elements=["foo", "bar"])
+        indexset_2 = run.optimization.indexsets.create("Indexset 2")
+        indexset_2.add([1, 2, 3])
+        # pandas can only convert dicts to dataframes if the values are lists
+        # or if index is given. But maybe using read_json instead of from_dict
+        # can remedy this. Or maybe we want to catch the resulting
+        # "ValueError: If using all scalar values, you must pass an index" and
+        # reraise a custom informative error?
+        test_data_1 = {"Indexset": ["foo"], "Indexset 2": [1]}
+        table = run.optimization.tables.create(
+            "Table",
+            constrained_to_indexsets=[indexset.name, indexset_2.name],
+        )
+        table.add(data=test_data_1)
+        assert table.data == test_data_1
+
+        table_2 = run.optimization.tables.create(
+            name="Table 2",
+            constrained_to_indexsets=[indexset.name, indexset_2.name],
+        )
+
+        with pytest.raises(ValueError, match="missing values"):
+            _ = table_2.add(
+                pd.DataFrame({"Indexset": [None], "Indexset 2": [2]}),
+                # empty string is allowed for now, but None or NaN raise
```

Comment:
can you please add a test for the empty string, just to be sure that this actually works?